lib/vfscore/syscalls.c

```diff
@@ -859,63 +859,48 @@ sys_rename(char *src, char *dest)
 int
 sys_symlink(const char *oldpath, const char *newpath)
 {
-	struct task	*t = main_task;
-	int		error;
-	char		op[PATH_MAX];
-	char		np[PATH_MAX];
-	struct dentry	*newdp;
-	struct dentry	*newdirdp;
-	char		*name;
-
-	if (oldpath == NULL || newpath == NULL) {
-		return (EFAULT);
-	}
+	int error;
+	char np[PATH_MAX];
+	struct dentry *newdp = NULL;
+	struct dentry *newdirdp = NULL;
+	char *name;
 
 	DPRINTF(VFSDB_SYSCALL, ("sys_link: oldpath=%s newpath=%s\n",
 				oldpath, newpath));
 
-	newdp		= NULL;
-	newdirdp	= NULL;
-
-	error = task_conv(t, newpath, VWRITE, np);
-	if (error != 0) {
-		return (error);
-	}
+	error = task_conv(main_task, newpath, VWRITE, np);
+	if (unlikely(error))
+		goto out;
 
-	/* parent directory for new path must exist */
-	if ((error = lookup(np, &newdirdp, &name)) != 0)
+	/* parent directory for newpath must exist */
+	error = lookup(np, &newdirdp, &name);
+	if (unlikely(error))
 		goto out;
 
 	vn_lock(newdirdp->d_vnode);
 
 	/* newpath should not already exist */
-	if (namei_last_nofollow(np, newdirdp, &newdp) == 0) {
+	if (unlikely(namei_last_nofollow(np, newdirdp, &newdp) == 0)) {
 		drele(newdp);
 		error = EEXIST;
-		goto out;
+		goto out_unlock;
 	}
 
+	UK_ASSERT(newdp == NULL);
```

Comment:
Nitpick, `checkpatch.pl` reports a `CHECK: Comparison to NULL could be written "!newdp"` on my end. Same thing, but perhaps worth considering. Does the CI/CD not run `checkpatch.pl` with the `--strict` flag?

Reply:
Thanks, changed it! 

No it is not using `--strict`.